packages/components/psammead-section-label/src/index.jsx

```diff
@@ -0,0 +1,147 @@
+import {
+  GEL_GROUP_3_SCREEN_WIDTH_MIN,
+  MEDIA_QUERY_TYPOGRAPHY,
+} from '@bbc/gel-foundations/breakpoints';
+import {
+  GEL_SPACING,
+  GEL_SPACING_DBL,
+  GEL_SPACING_TRPL,
+  GEL_SPACING_QUAD,
+} from '@bbc/gel-foundations/spacings';
+import {
+  getDoublePica,
+  GEL_FF_REITH_SANS,
+} from '@bbc/gel-foundations/typography';
+import { C_EBON, C_PEBBLE, C_WHITE } from '@bbc/psammead-styles/colours';
+
+const halfLineHeightRem = group => group.lineHeight / 2 / 16;
+
+const top = script => `
+  // place at middle of text line height
+  top: ${halfLineHeightRem(script.doublePica.groupA)}rem;
+
+  ${MEDIA_QUERY_TYPOGRAPHY.SMART_PHONE_ONLY} {
+    top: ${halfLineHeightRem(script.doublePica.groupB)}rem;
+  }
+
+  ${MEDIA_QUERY_TYPOGRAPHY.LAPTOP_AND_LARGER} {
+    top: ${halfLineHeightRem(script.doublePica.groupD)}rem;
+  }
+`;
+
+const WrapperWithoutBar = styled.div`
+  position: relative;
+  margin-top: ${GEL_SPACING_TRPL};
+  margin-bottom: ${GEL_SPACING_DBL};
+
+  ${MEDIA_QUERY_TYPOGRAPHY.LAPTOP_AND_LARGER} {
+    margin-top: ${GEL_SPACING_QUAD};
+    margin-bottom: ${GEL_SPACING_TRPL};
+  }
+`;
+
+const WrapperWithBar = styled(WrapperWithoutBar)`
+  &::before {
+    content: '';
+    position: absolute;
+    height: 0.0625rem;
+    border: none;
+    background: ${C_PEBBLE};
+    left: 0;
+    right: 0;
+    ${({ script }) => (script ? top(script) : 'top: 0')};
+  }
+`;
+
+const paddingDir = ({ dir }) => `padding-${dir === 'ltr' ? 'right' : 'left'}`;
+
+const Title = styled.h2`
+  ${({ script }) => script && getDoublePica(script)};
+  color: ${C_EBON};
+  background-color: ${C_WHITE};
+  font-family: ${GEL_FF_REITH_SANS};
+  font-weight: normal;
+  display: inline-block;
+  position: relative;
+
+  /* Unset the browser's default margins. */
+  margin: 0;
+
+  ${paddingDir}: ${GEL_SPACING};
+
+  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
+    ${paddingDir}: ${GEL_SPACING_DBL};
+  }
+`;
+
+Title.propTypes = {
+  dir: oneOf(['ltr', 'rtl']).isRequired,
+  script: shape(scriptPropType).isRequired,
+};
+
+const GenericSectionLabel = ({
+  children: text,
+  dir,
+  script,
+  wrapper: Wrapper,
+}) => (
+  // Only modify the Label to account for an inline title if there is a title to render.
+  <Wrapper script={text && script}>
+    {text && (
+      <Title script={script} dir={dir}>
+        {text}
+      </Title>
+    )}
+  </Wrapper>
+);
+
+GenericSectionLabel.defaultProps = {
+  children: null,
+};
+
+GenericSectionLabel.propTypes = {
+  children: string,
+  dir: oneOf(['ltr', 'rtl']).isRequired,
+  script: shape(scriptPropType).isRequired,
+  wrapper: oneOf([WrapperWithBar, WrapperWithoutBar]).isRequired,
+};
+
+export const SectionLabel = ({ children, dir, script }) => (
+  <GenericSectionLabel dir={dir} script={script} wrapper={WrapperWithBar}>
+    {children}
+  </GenericSectionLabel>
+);
+
+SectionLabel.defaultProps = {
+  children: null,
+  dir: 'ltr',
+  script: null,
+};
+
+SectionLabel.propTypes = {
+  children: string,
```

Comment:
Are `children` required?

Reply:
not in my implementation, I've handled the case where children aren't provided. My thinking was to handle the case where a strapline was missing from the data, and rendering the bar without the extra spacing for the missing text - https://github.com/bbc/psammead/pull/449/files#diff-a826825d0bceab40bf039858608d0224R56